lab-anthony/app/controllers/game-controller.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+//npm modules
+const angular = require('angular');
+
+//app modules
+const Monster = require('../models/monster.js');
+const Gem = require('../models/gem.js');
+
+//modules logic
+angular.module('monsterMash').controller('MonsterController', [MonsterController]);
+
+function MonsterController() {
+  this.history = ['You have entered the room! Get ready to mash some monsters!'];
+  this.moveCount = 0;
+  this.map = require('../models/map');
+
+  this.gameStart = false;
+  this.gameOver = false;
+
+  this.player = {
+    name: 'Big Money',
+    hp: 10,
+    damage: 10,
+    xp: 0,
+    location: 'roomA'
+  };
+
+  this.gameForm = {
+    direction: 'east'
+  };
+
+  this.room = {
+    name: this.player.location
+  };
+}
+
+MonsterController.prototype.moveDirection = function(direction) {
+  this.isDead();
+  this.moveCount ++;
+  var oldLocation = this.player.location;
+  var newLocation = this.map[oldLocation][direction];
+  if(newLocation && newLocation !== 'wall') {
+    this.updateLocation(newLocation);
+    return;
+  }
+
+  this.holdLocation();
+};
+
+MonsterController.prototype.updateLocation = function(location) {
+  this.player.location = location;
+  this.room.name = location;
+  if (Math.random() < 0.5) {
+    this.room.gem = new Gem();
+    this.player.hp += this.room.gem.addHp;
+    this.logTurn(`you found a Gem! It boosted your hp by ${this.room.gem.addHp}`);
+  }
+  if(Math.random() < this.map[location].monsterOdds) {
```

Comment:
It seems like this should check whether the room doesn't already have a monster too.